cmake_format/vscode_extension/src/extension.ts

```diff
@@ -55,43 +44,42 @@ export function activate(context: vscode.ExtensionContext) {
         console.log("Can't use cwd: " + cwd);
       }
 
-
       var env: any = {}
       if (config.get("mergeEnv", true)) {
         env = JSON.parse(JSON.stringify(process.env));
       }
+
       var cenv: any = config.get("env", {});
       if (cenv !== null) {
-        var delim = path.delimeter;
+        var delim = path.delimiter;
         for (var [key, value] of Object.entries(cenv)) {
           if (key.endsWith("PATH")) {
             var items = cenv[key].split(delim);
             if (key in env) {
-              items = items.concat(env[key].spit(delim));
+              items = items.concat(env[key].split(delim));
             }
             env[key] = items.join(delim);
           } else {
             env[key] = value;
           }
         }
       }
+
       if (!isEmpty(env)) {
         opts["env"] = env;
       }
 
-      const cp = require('child_process')
-
       // NOTE(josh): execFileSync will throw an Error if the
       // subprocess exits with non-zero status. The vscode GUI will
       // display the stacktrace in the corner.
-      var replacementText = cp.execFileSync(exePath, args, opts);
+      var replacementText = child_process.execFileSync(exePath, args, opts);
       var firstLine = document.lineAt(0);
       var lastLine = document.lineAt(document.lineCount - 1);
       var wholeRange = new vscode.Range(0,
         firstLine.range.start.character,
         document.lineCount - 1,
         lastLine.range.end.character);
-      return [vscode.TextEdit.replace(wholeRange, replacementText)];
+      return [vscode.TextEdit.replace(wholeRange, replacementText.toString())];
```

Comment:
why is `toString()` necessary?

Reply:
[TextEdit.replace](https://code.visualstudio.com/api/references/vscode-api#TextEdit) takes a `string` and [child_process.execFileSync](https://nodejs.org/api/child_process.html#child_process_child_process_execfilesync_file_args_options) was returning a `buffer` every time.

Reply:
Ok, thanks for the pointers. Seems like the conversion is proper, if not strictly necessary then?

Reply:
I don't know in what contexts it returns `string` though. And I'm not sure if there's some implicit conversion going on for `TextEdit.replace` if it *is* returning `buffer`. I know in the testing I did, if `child_process.execFileSync` returns `buffer` it causes an error. Something to be mindful of if there are any bug reports I suppose.